Improve installation

DanChadwick opened the pull request:
This PR:

- Allows opting-out of FullCalendar Scheduler.
- Specifies compatible semver versions of `fullcalendar` and `fullcalendar-scheduler` bower packages.
- Remove the automatic installation of `ember-cli-moment-shim` as it is not strictly needed.
- Updates the `README.md` to reflect the above changes.
- Adds a dependency for `ember-invoke-action`, which is required.

DanChadwick commented:
Sorry for the extra commits. I'm finding that pushing to github is the easiest way to test installation. This PR is *not* ready to commit yet.

jamesdixon commented:
Thanks, @DanChadwick. Just let me know when you feel that it's ready. Appreciate your contributions!

DanChadwick commented:
@jamesdixon: Depending upon the intention of the original code, it's ready now.

I removed a call to `this.addAddonToPackage` for `ember-cli-moment-shim` from `included() {}`. I believe the effect of this call was to modify the consuming app's `package.json` to include a dependency on `ember-cli-moment-shim`, which is not actually required. I don't understand however why `addAddonToPackage` as used rather than just an entry in `package.json / dependencies: {}`.

When I added the dependency for `ember-invoke-action` I did it by just adding an entry to `package.json / dependencies: {}`. The effect of this is to install `ember-invoke-action` but *not* update the consuming application's `package.json`. 

If the above is correct, please go ahead and merge this. I have installed it and believe it to be correct. Of course, it is very hard to test installation. :smile:

jamesdixon commented:
@DanChadwick I honestly don't recall what the purpose of `addAddonToPackage` was. I can't seem to find any documentation on it either. That said, I'll go ahead get this merged. I just merged the other two PRs and will cut a new version once this one is in place.

Thanks!

DanChadwick commented:
That's awesome. Yes, when I searched for `addAddonToPackage` I found an issue saying that it wasn't documented. I was amused.

jamesdixon commented on blueprints/ember-fullcalendar/index.js:
```diff
@@ -2,10 +2,8 @@ module.exports = {
-      return this.addBowerPackageToProject('fullcalendar-scheduler').then(() => {
-        return this.addAddonToProject('ember-cli-moment-shim', '0.6.2');
-      });
+    return this.addBowerPackageToProject('fullcalendar', '^2.7.3').then(() => {
```
@DanChadwick is it actually necessary to specify the version here? I assumed it would just pull whatever was specified in `bower.json`. If that is the case, this may be redundant. Otherwise, educate me :)

DanChadwick replied:
I was following the example of another module. Since you would normally specify some sort of version dependency when you install a package via `package.json` or `bower.json`, it seemed prudent to do so here. So for example if there were a version 2.0.0 of scheduler, which presumably is v2 because it has breaking changes, this would install an earlier version rather than the potentially breaking one.

Feel free to delete these if you think it's better. I was just trying to follow "best practices" -- whatever they might be.

jamesdixon replied:
Yeah, I suppose I may actually be a bit confused on how exactly packages are added. For example, in some addons, I see the bower packages in both `bower.json` and in the blueprint and in others it's just in the blueprint. Same thing for `package.json`.

As you mentioned, best practices seem hard to come by :) That said, I'm good with this.

DanChadwick replied:
Re bower. I did some research/asking yesterday. With bower dependencies, an addon's bower dependencies don't get installed into the `/bower_components` directory unless you explicitly do it with the blueprint. Then to actually include the code in `vendor.js` you do `app.import()`. The dependencies listed in `bower.json` are for your dummy app, not for users consuming your addon.

jamesdixon replied:
Ah! Thanks for the clarification -- super helpful!

jamesdixon commented:
All set. Thanks for the contribution!

DanChadwick commented:
Not at all. Thank *you* for your work as a maintainer.
